cupyx/fallback_mode/fallback.py

```diff
@@ -76,12 +100,239 @@ def __call__(self, *args, **kwargs):
             raise TypeError("'{}' object is not callable".format(
                 type(self._numpy_object).__name__))
 
+        # if ndarray method
+        if self._fallback_array is not None:
+            args = ((self._fallback_array,) + args)
+
         # Execute cupy method
         if self._cupy_object is not None:
-            return utils._call_cupy(self._cupy_object, args, kwargs)
+            return _call_cupy(self._cupy_object, args, kwargs)
 
         # Execute numpy method
-        return utils._call_numpy(self._numpy_object, args, kwargs)
+        return _call_numpy(self._numpy_object, args, kwargs)
 
 
 numpy = _RecursiveAttr(np, cp)
+
+# -----------------------------------------------------------------------------
+# utils
+# -----------------------------------------------------------------------------
+
+
+def _call_cupy(func, args, kwargs):
+    """
+    Calls cupy function with *args and **kwargs and
+    does necessary data transfers.
+
+    Args:
+        func: A cupy function that needs to be called.
+        args (tuple): Arguments.
+        kwargs (dict): Keyword arguments.
+
+    Returns:
+        Result after calling func and performing data transfers.
+    """
+
+    args, kwargs = _get_cupy_args(args, kwargs)
+
+    res = func(*args, **kwargs)
+
+    return _get_fallback_result(res)
+
+
+def _call_numpy(func, args, kwargs):
+    """
+    Calls numpy function with *args and **kwargs and
+    does necessary data transfers.
+
+    Args:
+        func: A numpy function that needs to be called.
+        args (tuple): Arguments.
+        kwargs (dict): Keyword arguments.
+
+    Returns:
+        Result after calling func and performing data transfers.
+    """
+
+    args, kwargs = _get_cupy_args(args, kwargs)
+
+    numpy_args, numpy_kwargs = _get_numpy_args(args, kwargs)
+
+    numpy_res = func(*numpy_args, **numpy_kwargs)
+
+    cupy_res = _get_cupy_result(numpy_res)
+
+    return _get_fallback_result(cupy_res)
+
+
+# -----------------------------------------------------------------------------
+# ndarray wrapper and proxy magic methods
+# -----------------------------------------------------------------------------
+
+
+class ndarray:
+    """
+    Wrapper around cupy.ndarray
+    Gets initialized with a cupy ndarray.
+    """
+
+    def __init__(self, array):
```

Comment:
Currently, this class is exposed to users, so it should have the same interface as `numpy.ndarray`.
```py
>>> import numpy
>>> type(numpy.array([1]))((2, 3), 'f')
array([[1.e-45, 0.e+00, 3.e-45],
       [0.e+00, 4.e-45, 0.e+00]], dtype=float32)

>>> from cupyx.fallback_mode import numpy
>>> type(numpy.array([1]))((2, 3), 'f')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __init__() takes 2 positional arguments but 3 were given
```

Reply:
```
>>> import numpy
>>> type(numpy.ndarray([1, 2, 3])) == numpy.ndarray
True
>>> from cupyx.fallback_mode import numpy
>>> type(numpy.ndarray([1, 2, 3])) == numpy.ndarray
False
```

Reply:
The problem arises due to `type(numpy.array([1]))` returning `fallback.ndarray`. Actual `numpy.ndarray` has same interface as that of `np.ndarray`.

Reply:
Okay, resolved!
And similar tests added.